### ISSUE TITLE
Go compile flags should be supported

### ISSUE THREAD

**Issue by privateducky** [preludeorg/build#299]
If I add this to the top of my test file:
```
// +build !windows
```
The test should compile for all but Windows. 

I would expect to get back only results from Linux/darwin.

**Comment by privateducky** [comment]
This change must occur in the Compute server.

**Comment by mireaulf** [comment]
We could catch build errors if we used fake filenames when compiling

**Comment by privateducky** [comment]
@mireaulf when you write a test with a build flag for only windows, does it only attempt the one platform?

**Comment by privateducky** [comment]
> We could catch build errors if we used fake filenames when compiling

Doesn't sound ideal.. do build flags work on current Go or is file name the only option?

**Comment by mireaulf** [comment]
Build flags works but they will be ignored if you specify the OS in the filename

**Comment by privateducky** [comment]
> Build flags works but they will be ignored if you specify the OS in the filename

At compile time, aren't the file names just uuid.go?

**Comment by mireaulf** [comment]
> > Build flags works but they will be ignored if you specify the OS in the filename
> 
> At compile time, aren't the file names just uuid.go?

That's correct but we do set the GOARCH env variable at compile time

**Comment by privateducky** [comment]
> > > Build flags works but they will be ignored if you specify the OS in the filename
> > 
> > 
> > At compile time, aren't the file names just uuid.go?
> 
> That's correct but we do set the GOARCH env variable at compile time

Not following. So if we set the GOARCH env variable and also have a build flag indicating only Windows.. do these conflict?

**Comment by privateducky** [comment]
Let me rephrase. I want to add this to a test file: 
```
// +build !windows
```
What do we need to do to support this?

**Comment by mireaulf** [comment]
> Let me rephrase. I want to add this to a test file:
> 
> ```
> // +build !windows
> ```
> 
> What do we need to do to support this?

I did some testing. As long as we specify the file we want to build `go build source.go`, it looks like the build constraints are completely ignored. We need a directory that contains the source file and a go.mod file (go.mod is requirement of `go build` without source file) 
<img width="562" alt="image" src="https://user-images.githubusercontent.com/7014725/213048789-ce2af3c6-5950-4a0b-8f97-fe27292ca12e.png">

**Comment by privateducky** [comment]
this ticket should be available on compute, not build. 
closing here.
